src/source/tcp.rs

```diff
@@ -108,36 +130,31 @@ impl Source for Int {
                     // TODO also add token_num here?
                     path: path.clone(), // captures server port
                 };
+                let tls_acceptor: Option<TlsAcceptor> = server_config
+                    .clone()
+                    .map(|s| TlsAcceptor::from(Arc::new(s)));
                 task::spawn(async move {
                     //let (reader, writer) = &mut (&stream, &stream);
-                    let mut buffer = [0; BUFFER_SIZE_BYTES];
                     if let Err(e) = tx.send(SourceReply::StartStream(stream_id)).await {
                         error!("TCP Error: {}", e);
                         return;
                     }
 
-                    while let Ok(n) = stream.read(&mut buffer).await {
-                        if n == 0 {
-                            if let Err(e) = tx.send(SourceReply::EndStream(stream_id)).await {
-                                error!("TCP Error: {}", e);
-                            };
-                            break;
-                        };
-                        if let Err(e) = tx
-                            .send(SourceReply::Data {
-                                origin_uri: origin_uri.clone(),
-                                // ALLOW: we define n as part of the read
-                                data: buffer[0..n].to_vec(),
-                                meta: None, // TODO: add peer address etc. to meta
-                                codec_override: None,
-                                stream: stream_id,
-                            })
-                            .await
-                        {
-                            error!("TCP Error: {}", e);
-                            break;
-                        };
-                    }
+                    if let Some(acceptor) = tls_acceptor {
+                        match acceptor.accept(stream).await {
+                            Ok(tls_stream) => {
+                                read_loop(tls_stream, tx, stream_id, origin_uri).await
+                            }
+                            Err(e) => {
```

Comment:
Dang, this `e` is not used.
```suggestion
                            Err(_e) => {
```